packages/ledger/src/index.ts

```diff
@@ -0,0 +1,206 @@
+import { Connector } from "@web3-react/types";
+
+import { LedgerConstructorArgs, LedgerOptions, LedgerProvider } from "./type";
+
+export const URI_AVAILABLE = "URI_AVAILABLE";
+
+function parseChainId(chainId: string | number) {
+  return typeof chainId === "string" ? Number.parseInt(chainId, 16) : chainId;
+}
+
+const isLogActive = false;
+
+export class Ledger extends Connector {
+  private readonly connectKitPromise = this.loadConnectKit();
+  private readonly defaultChainId: number = 1;
+  private readonly options: LedgerOptions;
+  private connectKit?: any;
+  public provider?: LedgerProvider;
+
+  constructor({ actions, options, onError }: LedgerConstructorArgs) {
+    isLogActive && console.log("Ledger Connector Constructor: Initializing...");
+    super(actions, onError);
+    this.options = options;
+  }
+
+  private chainChangedListener = (chainId: string): void => {
+    isLogActive &&
+      console.log("chainChangedListener: Handling chain changed event...");
+    this.actions.update({ chainId: Number.parseInt(chainId, 16) });
+  };
+
+  private accountsChangedListener = (accounts: string[]): void => {
+    isLogActive &&
+      console.log(
+        "accountsChangedListener: Handling accounts changed event..."
+      );
+    this.actions.update({ accounts });
+  };
+
+  private async isomorphicInitialize() {
+    console.group("isomorphicInitialize method");
+    isLogActive && console.log("isomorphicInitialize: Loading provider...");
+    try {
+      if (this.provider) return this.provider;
+      this.connectKit = await this.connectKitPromise;
+
+      const {
+        projectId,
+        chains,
+        optionalChains,
+        requiredMethods,
+        optionalMethods,
+        requiredEvents,
+        optionalEvents,
+        rpcMap = {
+          1: "https://cloudflare-eth.com/", // Mainnet
+          5: "https://goerli.optimism.io/", // Goerli
+          137: "https://polygon-rpc.com/", // Polygon
+        },
+      } = this.options;
+
+      this.connectKit.checkSupport({
+        providerType: "Ethereum",
+        walletConnectVersion: 2,
+        projectId,
+        chains,
+        optionalChains,
+        methods: requiredMethods,
+        optionalMethods,
+        events: requiredEvents,
+        optionalEvents,
+        rpcMap,
+      });
+      this.connectKit.enableDebugLogs();
+
+      const provider: LedgerProvider = (this.provider =
+        (await this.connectKit.getProvider()) as LedgerProvider);
+      provider.on("chainChanged", this.chainChangedListener);
+      provider.on("accountsChanged", this.accountsChangedListener);
+
+      return provider;
+    } finally {
+      console.groupEnd();
+    }
+  }
+
+  private async loadConnectKit() {
+    isLogActive && console.log("loadConnectKit: Loading Connect Kit...");
+
+    const src = "https://statuesque-naiad-0cb980.netlify.app/umd/index.js";
+    const globalName = "ledgerConnectKit";
```

Comment:
if you need a package you should load it via yarn

Reply:
This is an architecture decision that was made to enable hot upgrades on the connect kit.

Unfortunately, this is not a change we can implement at the moment.

Reply:
> if you need a package you should load it via yarn

Hero of the day sir, thanks for the common sense you showed on that sweet September 14

Reply:
This brings so much joy to my cold dead security heart that you spotted this and blocked merging because of it, saving many!